readthedocs/core/resolver.py

```diff
@@ -214,6 +214,58 @@ def resolve(
         )
         return urlunparse((protocol, domain, path, '', query_params, ''))
 
+    def _get_root_path(self, project):
+        """
+        Returns the root path for a project.
+
+        If the project is a subproject, it will return ``/projects/<project-slug>/``.
+        If the project is a main project, it will return ``/``.
+        This will respect the custom urlconf of the project if it's defined.
+        """
+        custom_prefix = None
+        if project.urlconf:
+            custom_prefix = project.urlconf.split("$", 1)[0]
```

Comment:
This could definitely use some explaining, or longer-form code that's more obvious what it's doing. It seems to be depending on `$var` names, and just pulling the path ahead of the first one? Seems brittle.

If we want this implementation to the be official one, we should likely make it a public function somewhere, instead of tucked into this private one.

Reply:
took this from https://github.com/readthedocs/readthedocs.org/blob/d641add9e66ee2937316ba3df7113bd7b17be884/readthedocs/projects/models.py#L633-L633

but +1 on a method